test/functional/legacy/011_autocommands_spec.lua

```diff
@@ -169,9 +171,15 @@ describe('file reading, writing and bufnew and filter autocommands', function()
        */]]))
     -- Need temp files here.
     feed_command('set shelltemp')
-    feed_command('au FilterReadPre   *.out  call rename(expand("<afile>"), expand("<afile>") . ".t")')
-    feed_command('au FilterReadPre   *.out  exe "silent !sed s/e/E/ " . shellescape(expand("<afile>")) . ".t >" . shellescape(expand("<afile>"))')
-    feed_command('au FilterReadPre   *.out  exe "silent !rm " . shellescape(expand("<afile>")) . ".t"')
+    if helpers.iswin() then
+      feed_command([[au FilterReadPre *.out exe 'silent !cat ' . ShellescPS1(expand('<afile>')) . " | \% {$_ -replace 'e','E'} > " . ShellescPS1(expand('<afile>') . '.t')]])
+      feed_command([[au FilterReadPre *.out exe '!silent cp ' . ShellescPS1(expand('<afile>') . '.t') ShellescPS1(expand('<afile>'))]])
+      feed_command([[au FilterReadPre *.out exe 'silent !rm ' . ShellescPS1(expand("<afile>") . '.t')]])
```

Comment:
This is the best workaround I can think of with cat.exe and without sed because powershell's ForEach-Object (aliased to %) acts on each line from cat.exe. Redirecting it to stdout will always empty the file first so the resulting file is empty or has the last line passed to ForEach-Object. To be safe, always use a tempfile for powershell output.

The test will fail if the output is encoding dependent because powershell uses utf-16 for redirection.